src/parser.cpp

```diff
@@ -15,6 +15,11 @@
 
 #include <cassert>
 
+static Token const& eof() {
+	static Token t = {TokenTag::END, "(EOF)", -1, -1, -1, -1};
+	return t;
+}
+
```

Comment:
This is intentionally duplicated I presume. Not a big deal but to keep in mind.

Reply:
Good catch. I forgot to delete the old one after doing the refactoring